tests/integration/server/nonclassevents/nonclassevent.controller.test.ts

```diff
@@ -0,0 +1,110 @@
+import { SinonStub, stub } from 'sinon';
+import { TestingModule, Test } from '@nestjs/testing';
+import { SessionModule } from 'nestjs-session';
+import { string } from 'testData';
+import { ConfigModule } from 'server/config/config.module';
+import { AuthModule } from 'server/auth/auth.module';
+import { AUTH_MODE } from 'common/constants';
+import { ConfigService } from 'server/config/config.service';
+import { strictEqual, deepStrictEqual } from 'assert';
+import request from 'supertest';
+import { NonClassEventService } from 'server/nonClassEvent/nonClassEvent.service';
+import { NonClassEventController } from 'server/nonClassEvent/nonClassEvent.controller';
+import {
+  computationalModelingofFluidsReadingGroup,
+  dataScienceReadingGroup,
+} from 'common/__tests__/data/nonClassEvents';
+import { NonClassParentView } from 'server/nonClassEvent/NonClassParentView.entity';
+import { TestingStrategy } from '../../../mocks/authentication/testing.strategy';
+
+const mockNonClassEventService = {
+  find: stub(),
+};
+
+describe('Non Class Event API', function () {
+  let authStub: SinonStub;
+  let api: HttpServer;
+
+  beforeEach(async function () {
+    authStub = stub(TestingStrategy.prototype, 'login');
+
+    const moduleRef: TestingModule = await Test.createTestingModule({
+      imports: [
+        SessionModule.forRoot({
+          session: {
+            secret: string,
+            resave: true,
+            saveUninitialized: true,
+          },
+        }),
+        ConfigModule,
+        AuthModule.register({
+          strategies: [TestingStrategy],
+          defaultStrategy: AUTH_MODE.TEST,
+        }),
+      ],
+      providers: [
+        {
+          provide: NonClassEventService,
+          useValue: mockNonClassEventService,
+        },
+      ],
+      controllers: [
+        NonClassEventController,
+      ],
+    })
+      .overrideProvider(ConfigService)
+      .useValue(new ConfigService({ NODE_ENV: 'development' }))
+
+      .compile();
+
+    const nestApp = await moduleRef.createNestApplication()
+      .init();
+
+    api = nestApp.getHttpServer() as HttpServer;
+  });
+  afterEach(function () {
+    authStub.restore();
+    Object.values(mockNonClassEventService)
+      .forEach((sinonStub: SinonStub): void => {
+        sinonStub.reset();
+      });
+  });
+  describe('GET /', function () {
+    describe('User is not authenticated', function () {
+      it('is inaccessible to unauthenticated users', async function () {
+        authStub.rejects(new ForbiddenException());
+
+        const response = await request(api).get('/api/non-class-events');
+
+        strictEqual(response.ok, false);
+        strictEqual(response.status, HttpStatus.FORBIDDEN);
+        strictEqual(mockNonClassEventService.find.callCount, 0);
+      });
+    });
+    describe('User is authenticated', function () {
+      it('it returns all the non-class events in the database', async function () {
+        authStub.resolves();
```

Comment:
Should this page also only be accessible by admin users only?

Reply:
Interesting question. I actually don't know. Add/Edit, certainly. View is a little less clear. I'll ask Chris.

Reply:
I just spoke to Chris about this and he said that all users should be able to _see_ the data. There will be some additional restrictions involved in adding/editing data - but that's another ticket.

As far as this ticket is concerned - all authenticated users should be able to see this data.